crates/ra_hir/src/code_model/src.rs

```diff
@@ -21,6 +22,13 @@ pub trait HasSource {
     fn source(self, db: &(impl DefDatabase + AstDatabase)) -> Source<Self::Ast>;
 }
 
+pub trait FromSource {
```

Comment:
Let's move this to a separate module, `from_source`. I think that, unlike `HasSource`, `FromSource` will be an approximate transformation, so we should clearly separate it from the rest of the hir